### PR TITLE
MAT-360: Log Donation ID and return 400 in case we get an attempt to …

### PR DESCRIPTION
**Pull request by bdsl**
…confirm a donation with no specified payment method ID

### CODE REVIEW

File: src/Application/Actions/Donations/Confirm.php [hunk 1]
```diff
@@ -92,6 +92,17 @@ protected function action(Request $request, Response $response, array $args): Re
             throw new NotFoundException();
         }
 
+        if (trim($paymentMethodId) === '') {
+            $donationID = $donation->getId();
+            $this->logger->error(
+                <<<EOF
+Donation Confirmation attempted with blank payment method id "$paymentMethodId" for Donation #$donationID
```
Comment (NoelLH) [inline src/Application/Actions/Donations/Confirm.php:99]: UUID is surfaced in more places so maybe more helpful to log instead / as well?

**Review by NoelLH — COMMENTED** [review]
Small suggestion on which ID we log, but looks good

File: src/Application/Actions/Donations/Confirm.php (hunk 1 above)
Comment (bdsl) [reply src/Application/Actions/Donations/Confirm.php:99]: Makes sense, I will change it.

Comment (bdsl) [reply src/Application/Actions/Donations/Confirm.php]: Done in 6900cdf

**Pull request merged**